ports/esp32s2/common-hal/busio/SPI.c

```diff
@@ -115,6 +115,16 @@ static void spi_bus_intr_disable(void *self)
 void common_hal_busio_spi_construct(busio_spi_obj_t *self,
         const mcu_pin_obj_t * clock, const mcu_pin_obj_t * mosi,
         const mcu_pin_obj_t * miso) {
+
+    //SCK is not optional. MOSI and MISO are
+    if (!clock) {
+        mp_raise_ValueError(translate("Must provide SCK pin"));
+    }
+
+    if (!miso && !mosi) {
+        mp_raise_ValueError(translate("Must provide MISO or MOSI pin"));
+    }
```

Comment:
shared-bindings should validate this because it applies to all implementations. Looks like it only checks clock, not MISO and MOSI. https://github.com/adafruit/circuitpython/blob/main/shared-bindings/busio/SPI.c#L95

Reply:
Ok, I can fix that.